mopidy_mpd/protocol/album_art.py

```diff
@@ -0,0 +1,107 @@
+from mopidy_mpd import protocol
+from urllib.request import urlopen
+
+
+def _get_art_url(self, uri):
+    images = self.core.library.get_images([uri]).get()
+    if images[uri]:
+        largest_image = sorted(
+            images[uri], key=lambda i: i.width or 0, reverse=True
+        )[0]
+        return largest_image.uri
+
+
+cover_cache = {}
+
+
+@protocol.commands.add("albumart")
+def albumart(context, uri, offset):
+    """
+    *musicpd.org, the music database section:*
+
+        ``albumart {URI} {OFFSET}``
+
+        Locate album art for the given song and return a chunk
+        of an album art image file at offset OFFSET.
+
+        This is currently implemented by searching the directory
+        the file resides in for a file called cover.png, cover.jpg,
+        cover.tiff or cover.bmp.
+
+        Returns the file size and actual number of bytes read at
+        the requested offset, followed by the chunk requested as
+        raw bytes (see Binary Responses), then a newline and the completion code.
+
+        Example::
+
+            albumart foo/bar.ogg 0
+            size: 1024768
+            binary: 8192
+            <8192 bytes>
+            OK
+
+    .. versionadded:: 0.21
+        New in MPD protocol version 0.21
+    """
+    global cover_cache
+
+    if uri not in cover_cache:
+        art_url = _get_art_url(context, uri)
+
+        if art_url is None:
+            return b"binary: 0\n"
```

Comment:
This is an error case, I'd have thought mpd provides a more descriptive error for something like this. And even if not, does it need to return binary in this case?

Reply:
Fixed in 12ca0fa